### PR TITLE
element-frame.lisp: poll the page instead of requiring user action

### PR DESCRIPTION
**Pull request by jmercouris**
I want the element frame results to update in the prompt buffer AS the user is dragging their selection. In order to achieve this I need to call `prompter::update` on some loop to invoke my `prompter:filter-preprocessor` which fetches the user selection.

Unfortunately, this does not work. I have to then press ANY key (not necessarily even in the prompt buffer), and then the sources update.

### CODE REVIEW

**Comment by jmercouris** [comment]
@Ambrevar Do you know how I should approach this?

**Comment by jmercouris** [comment]
Well, I did get a successful update to work doing:

```
(prompter::update-sources (current-prompt-buffer)) (run-thread (update-prompt-input (current-prompt-buffer)))
```

in a thread. Unfortunately, after doing so, the user couldn't scroll down because the results kept instantly updating. I think we'll need to fix the key dispatching per buffer before I can make a reasonable fix for the way that element-frame works.

**Comment by jmercouris** [comment]
Closing for now.

File: libraries/prompter/prompter-source.lisp [hunk 1]
```diff
@@ -342,7 +342,7 @@ interface.")
    (update-notifier (make-channel)
                     :type calispel:channel
                     :documentation "A channel which is written to when `filter'
-commits a change to `suggetsions'.  A notification is only send if at least
+commits a change to `suggestions'.  A notification is only sent if at least
```
Comment (Ambrevar) [inline libraries/prompter/prompter-source.lisp:345]: We could merge this :)

**Comment by Ambrevar** [comment]
```
(prompter::update-sources (current-prompt-buffer))
```
is correct.

However don't call `update-prompt-input` on something that gets updated in parallel because it updates _until_ there is no more changes.

If you just want to redraw the prompt, call `(prompt-render-suggestions prompt-buffer)`.

**Comment by Ambrevar** [comment]
Or `prompt-render` to redraw the entire prompt and not just the suggestions.

**Pull request closed without merging**